drivers/hmis/app/graphql/mutations/delete_enrollment.rb

```diff
@@ -13,14 +13,24 @@ class DeleteEnrollment < BaseMutation
     def resolve(id:)
       enrollment = Hmis::Hud::Enrollment.viewable_by(current_user).find_by(id: id)
       raise HmisErrors::ApiError, 'Record not found' unless enrollment.present?
-      # WIP Enrollments can be deleted if user has "can_edit_enrollments" access for this project
-      raise HmisErrors::ApiError, 'Access denied' unless current_user.permissions_for?(enrollment, :can_edit_enrollments)
 
       errors = []
       if enrollment.in_progress?
+        # WIP Enrollments can be deleted if user has "can_edit_enrollments" access for this project
+        raise HmisErrors::ApiError, 'Access denied' unless current_user.permissions_for?(enrollment, :can_edit_enrollments)
+
+        enrollment.destroy!
+      elsif !enrollment.intake_assessment
+        # Non-WIP, Active Enrollments can be destroyed if there is no associated intake assessment
+        # (either due to data being migrated in, or if we support projects that don't do assesments in future).
+        raise HmisErrors::ApiError, 'Access denied' unless current_user.permissions_for?(enrollment, :can_delete_enrollments)
+
         enrollment.destroy!
       else
-        # Deleting non-WIP Enrollments requires can_delete_enrollments, and can only occur via DeleteAssessment mutation (deleting intake)
+        # WIP Enrollments can be deleted if user has "can_edit_enrollments" access for this project
+        raise HmisErrors::ApiError, 'Access denied' unless current_user.permissions_for?(enrollment, :can_edit_enrollments)
+
+        # Deleting non-WIP Enrollements requires can_delete_enrollments, and can only occur via DeleteAssessment mutation (deleting intake)
         errors << HmisErrors::Error.new(:base, full_message: 'Completed enrollments can not be deleted. Please exit the client instead.')
       end
```

Comment:
This logic is sound, but I think it could be cleaned up a little. The user really should have `can_edit_enrollments` in all 3 cases. (There is no scenario where someone should be able to delete enrollments but not edit them.) So I wonder, rather than having the can_edit_enrollments check on both lines 20 and 30, if we should just keep it on line 17 where it was. That way we would _always_ ensure the user has `can_edit_enrollments` before proceeding.

The comment, `WIP Enrollments can be deleted if user has "can_edit_enrollments" access for this project`, only really makes sense to stay on line 19 I think -- since that is where we are actually handling the WIP Enrollment case.

Reply:
Thanks Gig, perfect!